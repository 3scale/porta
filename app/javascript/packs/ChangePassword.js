import { ChangePasswordWrapper as ChangePassword } from 'ChangePassword'
import { safeFromJsonString } from 'utilities'

document.addEventListener('DOMContentLoaded', () => {
  const changePasswordContainer = document.getElementById('pf-login-page-container')
  const changePasswordProps = safeFromJsonString(changePasswordContainer.dataset.changePasswordProps)
  ChangePassword(changePasswordProps, 'pf-login-page-container')
})
