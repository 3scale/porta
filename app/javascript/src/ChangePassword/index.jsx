// @flow

export * from 'ChangePassword/components/ChangePassword'
export * from 'ChangePassword/components/ChangePasswordHooks'
