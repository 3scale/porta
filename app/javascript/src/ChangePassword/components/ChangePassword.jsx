// @flow

import * as React from 'react'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { LoginPage, Form, ActionGroup, Button } from '@patternfly/react-core'
import { PasswordField, PasswordConfirmationField, HiddenInputs, FlashMessages } from 'LoginPage'
import { useFormState } from 'ChangePassword'
import type { FlashMessage } from 'Types'

import 'LoginPage/assets/styles/loginPage.scss'
import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'

type Props = {
  lostPasswordToken: ?string,
  url: string,
  errors: (?FlashMessage)[]
}

const ChangePassword = ({ lostPasswordToken, url, errors }: Props): React.Node => {
  const {
    isFormDisabled,
    onFormChange,
    password,
    passwordConfirmation
  } = useFormState()

  const passwordProps = {
    isRequired: true,
    name: 'user[password]',
    fieldId: 'user_password',
    label: 'Password',
    value: password.value,
    isValid: password.isValid,
    errorMessage: password.errorMessage,
    autoFocus: 'autoFocus',
    onBlur: password.onBlur,
    onChange: password.onChange
  }

  const passwordConfirmationProps = {
    isRequired: true,
    name: 'user[password_confirmation]',
    fieldId: 'user_password_confirmation',
    label: 'Password confirmation',
    value: passwordConfirmation.value,
    isValid: passwordConfirmation.isValid,
    errorMessage: passwordConfirmation.errorMessage,
    onBlur: passwordConfirmation.onBlur,
    onChange: passwordConfirmation.onChange
  }

  return (
    <LoginPage
      brandImgSrc={brandImg}
      brandImgAlt='Red Hat 3scale API Management'
      backgroundImgSrc={PF4DownstreamBG}
      backgroundImgAlt='Red Hat 3scale API Management'
      loginTitle='Change Password'
      footer={null}
    >
      {errors && <FlashMessages flashMessages={errors}/>}
      <Form
        action={url}
        noValidate
        id='edit_user_2'
        acceptCharset='UTF-8'
        method='post'
        onChange={onFormChange}
      >
        <input type='hidden' name='_method' value='put' />
        <HiddenInputs />
        <PasswordField inputProps={passwordProps} />
        <PasswordConfirmationField inputProps={passwordConfirmationProps} />
        {lostPasswordToken &&
          <input id='password_reset_token' type='hidden' name='password_reset_token' value={lostPasswordToken} />
        }
        <ActionGroup>
          <Button
            className='pf-c-button pf-m-primary pf-m-block'
            type='submit'
            name='commit'
            isDisabled={isFormDisabled}
          >
            Change Password
          </Button>
        </ActionGroup>
      </Form>
    </LoginPage>
  )
}

const ChangePasswordWrapper = (props: Props, containerId: string): void =>
  createReactWrapper(<ChangePassword {...props} />, containerId)

export { ChangePassword, ChangePasswordWrapper }
