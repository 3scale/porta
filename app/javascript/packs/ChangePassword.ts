import { ChangePasswordWrapper } from 'ChangePassword/components/ChangePassword'
import { safeFromJsonString } from 'utilities/json-utils'

import type { Props } from 'ChangePassword/components/ChangePassword'

document.addEventListener('DOMContentLoaded', () => {
  const containerId = 'pf-login-page-container'
  const changePasswordContainer = document.getElementById(containerId)

  if (!changePasswordContainer) {
    throw new Error('The target ID was not found: ' + containerId)
  }

  const { lostPasswordToken = null, url = '', errors = [] } = safeFromJsonString<Props>(changePasswordContainer.dataset.changePasswordProps) as Props

  ChangePasswordWrapper({
    lostPasswordToken,
    url,
    errors
  }, 'pf-login-page-container')
})
