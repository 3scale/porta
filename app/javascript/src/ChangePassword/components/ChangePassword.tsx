import { createReactWrapper } from 'utilities/createReactWrapper'
import { ActionGroup, Button, Form, LoginPage } from '@patternfly/react-core'
import { useFormState } from 'ChangePassword/components/ChangePasswordHooks'
import { FlashMessages } from 'LoginPage/loginForms/FlashMessages'
import { PasswordField, PasswordConfirmationField } from 'LoginPage/loginForms/FormGroups'
import { HiddenInputs } from 'LoginPage/loginForms/HiddenInputs'
import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'
import 'LoginPage/assets/styles/loginPage.scss'

import type { FlashMessage, InputProps } from 'Types'
import type { FunctionComponent } from 'react'

type Props = {
  lostPasswordToken: string | null | undefined,
  url: string,
  errors?: FlashMessage[]
}

const ChangePassword: FunctionComponent<Props> = ({
  lostPasswordToken,
  url,
  errors
}) => {
  const {
    isFormDisabled,
    onFormChange,
    password,
    passwordConfirmation
  } = useFormState()

  const passwordProps: InputProps = {
    isRequired: true,
    name: 'user[password]',
    fieldId: 'user_password',
    label: 'Password',
    value: password.value,
    isValid: password.isValid,
    errorMessage: password.errorMessage,
    autoFocus: true,
    onBlur: password.onBlur,
    onChange: password.onChange
  }

  const passwordConfirmationProps = {
    isRequired: true,
    name: 'user[password_confirmation]',
    fieldId: 'user_password_confirmation',
    label: 'Password confirmation',
    value: passwordConfirmation.value,
    isValid: passwordConfirmation.isValid,
    errorMessage: passwordConfirmation.errorMessage,
    onBlur: passwordConfirmation.onBlur,
    onChange: passwordConfirmation.onChange
  }

  return (
    <LoginPage
      backgroundImgAlt='Red Hat 3scale API Management'
      backgroundImgSrc={PF4DownstreamBG}
      brandImgAlt='Red Hat 3scale API Management'
      brandImgSrc={brandImg}
      loginTitle='Change Password'
    >
      {errors && <FlashMessages flashMessages={errors} />}
      <Form
        noValidate
        acceptCharset='UTF-8'
        action={url}
        id='edit_user_2'
        method='post'
        onChange={onFormChange}
      >
        <input name='_method' type='hidden' value='put' />
        <HiddenInputs />
        <PasswordField inputProps={passwordProps} />
        <PasswordConfirmationField inputProps={passwordConfirmationProps} />
        {lostPasswordToken ? <input id='password_reset_token' name='password_reset_token' type='hidden' value={lostPasswordToken} /> : null}
        <ActionGroup>
          <Button
            className='pf-c-button pf-m-primary pf-m-block'
            isDisabled={isFormDisabled}
            name='commit'
            type='submit'
          >
          Change Password
          </Button>
        </ActionGroup>
      </Form>
    </LoginPage>
  )
}

// eslint-disable-next-line react/jsx-props-no-spreading
const ChangePasswordWrapper = (props: Props, containerId: string): void => createReactWrapper(<ChangePassword {...props} />, containerId)

export { ChangePassword, ChangePasswordWrapper }
