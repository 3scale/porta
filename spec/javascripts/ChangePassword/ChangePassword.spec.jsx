// @flow

import React from 'react'
import { mount } from 'enzyme'
import { ChangePassword } from 'ChangePassword'

it('should render itself', () => {
  const props = {
    lostPasswordToken: 'foo',
    url: 'foo/bar',
    errors: []
  }
  const wrapper = mount(<ChangePassword {...props} />)
  expect(wrapper).toMatchSnapshot()
})

it('should render with server side errors when present', () => {
  const props = {
    lostPasswordToken: 'foo',
    url: 'foo/bar',
    errors: [{type: 'error', message: 'Ooops!'}]
  }
  const wrapper = mount(<ChangePassword {...props} />)
  expect(wrapper).toMatchSnapshot()
})

// TODO: Test Form errors and interaction.
