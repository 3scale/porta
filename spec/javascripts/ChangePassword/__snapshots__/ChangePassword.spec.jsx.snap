// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<ChangePassword
  errors={Array []}
  lostPasswordToken="foo"
  url="foo/bar"
>
  <LoginPage
    backgroundImgAlt="Red Hat 3scale API Management"
    backgroundImgSrc=""
    brandImgAlt="Red Hat 3scale API Management"
    brandImgSrc=""
    footer={null}
    loginTitle="Change Password"
  >
    <Login
      className=""
      footer={null}
      header={
        <LoginHeader
          headerBrand={
            <React.Fragment>
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              />
            </React.Fragment>
          }
        />
      }
    >
      <div
        className="pf-c-login"
      >
        <div
          className="pf-c-login__container"
        >
          <LoginHeader
            headerBrand={
              <React.Fragment>
                <Brand
                  alt="Red Hat 3scale API Management"
                  src=""
                />
              </React.Fragment>
            }
          >
            <header
              className="pf-c-login__header"
            >
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              >
                <img
                  alt="Red Hat 3scale API Management"
                  className="pf-c-brand"
                  src=""
                />
              </Brand>
            </header>
          </LoginHeader>
          <main
            className="pf-c-login__main"
          >
            <LoginMainHeader
              title="Change Password"
            >
              <header
                className="pf-c-login__main-header"
              >
                <Title
                  headingLevel="h2"
                  size="3xl"
                >
                  <h2
                    className="pf-c-title pf-m-3xl"
                  >
                    Change Password
                  </h2>
                </Title>
              </header>
            </LoginMainHeader>
            <LoginMainBody>
              <div
                className="pf-c-login__main-body"
              >
                <FlashMessages
                  flashMessages={Array []}
                >
                  <div />
                </FlashMessages>
                <Form
                  acceptCharset="UTF-8"
                  action="foo/bar"
                  id="edit_user_2"
                  method="post"
                  noValidate={true}
                  onChange={[Function]}
                >
                  <form
                    acceptCharset="UTF-8"
                    action="foo/bar"
                    className="pf-c-form"
                    id="edit_user_2"
                    method="post"
                    noValidate={true}
                    onChange={[Function]}
                  >
                    <input
                      name="_method"
                      type="hidden"
                      value="put"
                    />
                    <HiddenInputs>
                      <input
                        name="utf8"
                        type="hidden"
                        value="âœ“"
                      />
                      <CSRFToken>
                        <input
                          type="hidden"
                        />
                      </CSRFToken>
                    </HiddenInputs>
                    <PasswordField
                      inputProps={
                        Object {
                          "autoFocus": "autoFocus",
                          "errorMessage": undefined,
                          "fieldId": "user_password",
                          "isRequired": true,
                          "isValid": true,
                          "label": "Password",
                          "name": "user[password]",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "value": "",
                        }
                      }
                    >
                      <FormGroup
                        helperTextInvalid="Password is mandatory"
                        inputProps={
                          Object {
                            "autoFocus": "autoFocus",
                            "errorMessage": undefined,
                            "fieldId": "user_password",
                            "isRequired": true,
                            "isValid": true,
                            "label": "Password",
                            "name": "user[password]",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "value": "",
                          }
                        }
                        type="password"
                      >
                        <FormGroup
                          fieldId="user_password"
                          helperTextInvalid="Password is mandatory"
                          isRequired={true}
                          isValid={true}
                          label="Password"
                        >
                          <div
                            className="pf-c-form__group"
                          >
                            <label
                              className="pf-c-form__label"
                              htmlFor="user_password"
                            >
                              <span
                                className="pf-c-form__label-text"
                              >
                                Password
                              </span>
                              <span
                                aria-hidden="true"
                                className="pf-c-form__label-required"
                              >
                                *
                              </span>
                            </label>
                            <ForwardRef
                              autoFocus="autoFocus"
                              id="user_password"
                              isRequired={true}
                              isValid={true}
                              name="user[password]"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="password"
                              value=""
                            >
                              <TextInputBase
                                aria-label={null}
                                autoFocus="autoFocus"
                                className=""
                                id="user_password"
                                innerRef={null}
                                isDisabled={false}
                                isReadOnly={false}
                                isRequired={true}
                                isValid={true}
                                name="user[password]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <input
                                  aria-invalid={false}
                                  aria-label={null}
                                  autoFocus="autoFocus"
                                  className="pf-c-form-control"
                                  disabled={false}
                                  id="user_password"
                                  name="user[password]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  readOnly={false}
                                  required={true}
                                  type="password"
                                  value=""
                                />
                              </TextInputBase>
                            </ForwardRef>
                          </div>
                        </FormGroup>
                      </FormGroup>
                    </PasswordField>
                    <PasswordConfirmationField
                      inputProps={
                        Object {
                          "errorMessage": undefined,
                          "fieldId": "user_password_confirmation",
                          "isRequired": true,
                          "isValid": true,
                          "label": "Password confirmation",
                          "name": "user[password_confirmation]",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "value": "",
                        }
                      }
                    >
                      <FormGroup
                        helperTextInvalid="Password confirmation is mandatory"
                        inputProps={
                          Object {
                            "errorMessage": undefined,
                            "fieldId": "user_password_confirmation",
                            "isRequired": true,
                            "isValid": true,
                            "label": "Password confirmation",
                            "name": "user[password_confirmation]",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "value": "",
                          }
                        }
                        type="password"
                      >
                        <FormGroup
                          fieldId="user_password_confirmation"
                          helperTextInvalid="Password confirmation is mandatory"
                          isRequired={true}
                          isValid={true}
                          label="Password confirmation"
                        >
                          <div
                            className="pf-c-form__group"
                          >
                            <label
                              className="pf-c-form__label"
                              htmlFor="user_password_confirmation"
                            >
                              <span
                                className="pf-c-form__label-text"
                              >
                                Password confirmation
                              </span>
                              <span
                                aria-hidden="true"
                                className="pf-c-form__label-required"
                              >
                                *
                              </span>
                            </label>
                            <ForwardRef
                              id="user_password_confirmation"
                              isRequired={true}
                              isValid={true}
                              name="user[password_confirmation]"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="password"
                              value=""
                            >
                              <TextInputBase
                                aria-label={null}
                                className=""
                                id="user_password_confirmation"
                                innerRef={null}
                                isDisabled={false}
                                isReadOnly={false}
                                isRequired={true}
                                isValid={true}
                                name="user[password_confirmation]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <input
                                  aria-invalid={false}
                                  aria-label={null}
                                  className="pf-c-form-control"
                                  disabled={false}
                                  id="user_password_confirmation"
                                  name="user[password_confirmation]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  readOnly={false}
                                  required={true}
                                  type="password"
                                  value=""
                                />
                              </TextInputBase>
                            </ForwardRef>
                          </div>
                        </FormGroup>
                      </FormGroup>
                    </PasswordConfirmationField>
                    <input
                      id="password_reset_token"
                      name="password_reset_token"
                      type="hidden"
                      value="foo"
                    />
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Component
                            className="pf-c-button pf-m-primary pf-m-block"
                            isDisabled={true}
                            name="commit"
                            type="submit"
                          >
                            <ComponentWithOuia
                              component={[Function]}
                              componentProps={
                                Object {
                                  "children": "Change Password",
                                  "className": "pf-c-button pf-m-primary pf-m-block",
                                  "isDisabled": true,
                                  "name": "commit",
                                  "type": "submit",
                                }
                              }
                              consumerContext={null}
                            >
                              <Button
                                className="pf-c-button pf-m-primary pf-m-block"
                                isDisabled={true}
                                name="commit"
                                ouiaContext={
                                  Object {
                                    "isOuia": false,
                                    "ouiaId": null,
                                  }
                                }
                                type="submit"
                              >
                                <button
                                  aria-disabled={null}
                                  aria-label={null}
                                  className="pf-c-button pf-m-primary pf-c-button pf-m-primary pf-m-block"
                                  disabled={true}
                                  name="commit"
                                  tabIndex={null}
                                  type="submit"
                                >
                                  Change Password
                                </button>
                              </Button>
                            </ComponentWithOuia>
                          </Component>
                        </div>
                      </div>
                    </ActionGroup>
                  </form>
                </Form>
              </div>
            </LoginMainBody>
          </main>
        </div>
      </div>
    </Login>
  </LoginPage>
</ChangePassword>
`;

exports[`should render with server side errors when present 1`] = `
<ChangePassword
  errors={
    Array [
      Object {
        "message": "Ooops!",
        "type": "error",
      },
    ]
  }
  lostPasswordToken="foo"
  url="foo/bar"
>
  <LoginPage
    backgroundImgAlt="Red Hat 3scale API Management"
    backgroundImgSrc=""
    brandImgAlt="Red Hat 3scale API Management"
    brandImgSrc=""
    footer={null}
    loginTitle="Change Password"
  >
    <Login
      className=""
      footer={null}
      header={
        <LoginHeader
          headerBrand={
            <React.Fragment>
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              />
            </React.Fragment>
          }
        />
      }
    >
      <div
        className="pf-c-login"
      >
        <div
          className="pf-c-login__container"
        >
          <LoginHeader
            headerBrand={
              <React.Fragment>
                <Brand
                  alt="Red Hat 3scale API Management"
                  src=""
                />
              </React.Fragment>
            }
          >
            <header
              className="pf-c-login__header"
            >
              <Brand
                alt="Red Hat 3scale API Management"
                src=""
              >
                <img
                  alt="Red Hat 3scale API Management"
                  className="pf-c-brand"
                  src=""
                />
              </Brand>
            </header>
          </LoginHeader>
          <main
            className="pf-c-login__main"
          >
            <LoginMainHeader
              title="Change Password"
            >
              <header
                className="pf-c-login__main-header"
              >
                <Title
                  headingLevel="h2"
                  size="3xl"
                >
                  <h2
                    className="pf-c-title pf-m-3xl"
                  >
                    Change Password
                  </h2>
                </Title>
              </header>
            </LoginMainHeader>
            <LoginMainBody>
              <div
                className="pf-c-login__main-body"
              >
                <FlashMessages
                  flashMessages={
                    Array [
                      Object {
                        "message": "Ooops!",
                        "type": "error",
                      },
                    ]
                  }
                >
                  <div>
                    <p
                      className="pf-c-form__helper-text pf-m-error"
                      key="Ooops!"
                    >
                      <ExclamationCircleIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                        title={null}
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                            transform=""
                          />
                        </svg>
                      </ExclamationCircleIcon>
                       Ooops!
                    </p>
                  </div>
                </FlashMessages>
                <Form
                  acceptCharset="UTF-8"
                  action="foo/bar"
                  id="edit_user_2"
                  method="post"
                  noValidate={true}
                  onChange={[Function]}
                >
                  <form
                    acceptCharset="UTF-8"
                    action="foo/bar"
                    className="pf-c-form"
                    id="edit_user_2"
                    method="post"
                    noValidate={true}
                    onChange={[Function]}
                  >
                    <input
                      name="_method"
                      type="hidden"
                      value="put"
                    />
                    <HiddenInputs>
                      <input
                        name="utf8"
                        type="hidden"
                        value="âœ“"
                      />
                      <CSRFToken>
                        <input
                          type="hidden"
                        />
                      </CSRFToken>
                    </HiddenInputs>
                    <PasswordField
                      inputProps={
                        Object {
                          "autoFocus": "autoFocus",
                          "errorMessage": undefined,
                          "fieldId": "user_password",
                          "isRequired": true,
                          "isValid": true,
                          "label": "Password",
                          "name": "user[password]",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "value": "",
                        }
                      }
                    >
                      <FormGroup
                        helperTextInvalid="Password is mandatory"
                        inputProps={
                          Object {
                            "autoFocus": "autoFocus",
                            "errorMessage": undefined,
                            "fieldId": "user_password",
                            "isRequired": true,
                            "isValid": true,
                            "label": "Password",
                            "name": "user[password]",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "value": "",
                          }
                        }
                        type="password"
                      >
                        <FormGroup
                          fieldId="user_password"
                          helperTextInvalid="Password is mandatory"
                          isRequired={true}
                          isValid={true}
                          label="Password"
                        >
                          <div
                            className="pf-c-form__group"
                          >
                            <label
                              className="pf-c-form__label"
                              htmlFor="user_password"
                            >
                              <span
                                className="pf-c-form__label-text"
                              >
                                Password
                              </span>
                              <span
                                aria-hidden="true"
                                className="pf-c-form__label-required"
                              >
                                *
                              </span>
                            </label>
                            <ForwardRef
                              autoFocus="autoFocus"
                              id="user_password"
                              isRequired={true}
                              isValid={true}
                              name="user[password]"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="password"
                              value=""
                            >
                              <TextInputBase
                                aria-label={null}
                                autoFocus="autoFocus"
                                className=""
                                id="user_password"
                                innerRef={null}
                                isDisabled={false}
                                isReadOnly={false}
                                isRequired={true}
                                isValid={true}
                                name="user[password]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <input
                                  aria-invalid={false}
                                  aria-label={null}
                                  autoFocus="autoFocus"
                                  className="pf-c-form-control"
                                  disabled={false}
                                  id="user_password"
                                  name="user[password]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  readOnly={false}
                                  required={true}
                                  type="password"
                                  value=""
                                />
                              </TextInputBase>
                            </ForwardRef>
                          </div>
                        </FormGroup>
                      </FormGroup>
                    </PasswordField>
                    <PasswordConfirmationField
                      inputProps={
                        Object {
                          "errorMessage": undefined,
                          "fieldId": "user_password_confirmation",
                          "isRequired": true,
                          "isValid": true,
                          "label": "Password confirmation",
                          "name": "user[password_confirmation]",
                          "onBlur": [Function],
                          "onChange": [Function],
                          "value": "",
                        }
                      }
                    >
                      <FormGroup
                        helperTextInvalid="Password confirmation is mandatory"
                        inputProps={
                          Object {
                            "errorMessage": undefined,
                            "fieldId": "user_password_confirmation",
                            "isRequired": true,
                            "isValid": true,
                            "label": "Password confirmation",
                            "name": "user[password_confirmation]",
                            "onBlur": [Function],
                            "onChange": [Function],
                            "value": "",
                          }
                        }
                        type="password"
                      >
                        <FormGroup
                          fieldId="user_password_confirmation"
                          helperTextInvalid="Password confirmation is mandatory"
                          isRequired={true}
                          isValid={true}
                          label="Password confirmation"
                        >
                          <div
                            className="pf-c-form__group"
                          >
                            <label
                              className="pf-c-form__label"
                              htmlFor="user_password_confirmation"
                            >
                              <span
                                className="pf-c-form__label-text"
                              >
                                Password confirmation
                              </span>
                              <span
                                aria-hidden="true"
                                className="pf-c-form__label-required"
                              >
                                *
                              </span>
                            </label>
                            <ForwardRef
                              id="user_password_confirmation"
                              isRequired={true}
                              isValid={true}
                              name="user[password_confirmation]"
                              onBlur={[Function]}
                              onChange={[Function]}
                              type="password"
                              value=""
                            >
                              <TextInputBase
                                aria-label={null}
                                className=""
                                id="user_password_confirmation"
                                innerRef={null}
                                isDisabled={false}
                                isReadOnly={false}
                                isRequired={true}
                                isValid={true}
                                name="user[password_confirmation]"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="password"
                                validated="default"
                                value=""
                              >
                                <input
                                  aria-invalid={false}
                                  aria-label={null}
                                  className="pf-c-form-control"
                                  disabled={false}
                                  id="user_password_confirmation"
                                  name="user[password_confirmation]"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  readOnly={false}
                                  required={true}
                                  type="password"
                                  value=""
                                />
                              </TextInputBase>
                            </ForwardRef>
                          </div>
                        </FormGroup>
                      </FormGroup>
                    </PasswordConfirmationField>
                    <input
                      id="password_reset_token"
                      name="password_reset_token"
                      type="hidden"
                      value="foo"
                    />
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Component
                            className="pf-c-button pf-m-primary pf-m-block"
                            isDisabled={true}
                            name="commit"
                            type="submit"
                          >
                            <ComponentWithOuia
                              component={[Function]}
                              componentProps={
                                Object {
                                  "children": "Change Password",
                                  "className": "pf-c-button pf-m-primary pf-m-block",
                                  "isDisabled": true,
                                  "name": "commit",
                                  "type": "submit",
                                }
                              }
                              consumerContext={null}
                            >
                              <Button
                                className="pf-c-button pf-m-primary pf-m-block"
                                isDisabled={true}
                                name="commit"
                                ouiaContext={
                                  Object {
                                    "isOuia": false,
                                    "ouiaId": null,
                                  }
                                }
                                type="submit"
                              >
                                <button
                                  aria-disabled={null}
                                  aria-label={null}
                                  className="pf-c-button pf-m-primary pf-c-button pf-m-primary pf-m-block"
                                  disabled={true}
                                  name="commit"
                                  tabIndex={null}
                                  type="submit"
                                >
                                  Change Password
                                </button>
                              </Button>
                            </ComponentWithOuia>
                          </Component>
                        </div>
                      </div>
                    </ActionGroup>
                  </form>
                </Form>
              </div>
            </LoginMainBody>
          </main>
        </div>
      </div>
    </Login>
  </LoginPage>
</ChangePassword>
`;
